src/Main.hs

```diff
@@ -145,16 +148,17 @@ lookupNode :: ModName -> NodesC -> Maybe Int
 lookupNode ([],m) (Trie.Sub _ mb) = lookupBy (containsModule m) =<< mb
 
 lookupNode (q:qs,m) (Trie.Sub ts mb) =
-  (lookupBy (isCollapsed q) =<< mb) `mplus`
+  (lookupBy containsNode =<< mb) `mplus`
   (lookupNode (qs,m) =<< Map.lookup q ts)
 
-containsModule :: String -> (NodeT, String) -> Bool
-containsModule q (ModuleNode, q1)             = q == q1
-containsModule q (CollapsedNode withMod, q1)  = withMod && q == q1
+  where
+  containsNode :: (NodeT, String) -> Bool
+  containsNode (CollapsedNode _ t, q1) = q == q1 && isJust (lookupNode (qs,m) t)
+  containsNode (ModuleNode, _)         = False
 
-isCollapsed :: String -> (NodeT, String) -> Bool
-isCollapsed q (CollapsedNode _, q1) = q == q1
-isCollapsed _ (ModuleNode, _)       = False
+containsModule :: String -> (NodeT, String) -> Bool
```

Comment:
`containsModule` didn't change; it's just an innocent victim of `isCollapsed` being moved and renamed.